### PR TITLE
feat: add CLI support

### PR DESCRIPTION
**Pull request by xgaozoyoe**
adding CLI support to play with the wasm ZK emulator.

### CODE REVIEW

File: src/cli/main.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        .value_parser(value_parser!(std::string::String));
+    let fn_name_arg = Arg::with_name("function_name")
+        .long("fname")
+        .short('f')
+        .required(true)
+        .value_name("FUNCTIONNAME")
+        .help("Function you would like to run from the file")
+        .takes_value(true)
+        .value_parser(value_parser!(std::string::String));
+    let type_arg = Arg::with_name("types")
```
Comment (junyu0312) [inline src/cli/main.rs:27]: We can fetch types from interpreter, for example,
```
        if let Some(ExternVal::Func(func)) = compile_outcome.instance.export_by_name("test") {
            println!("{:?}", func.signature());
        }
```

File: src/circuits/mod.rs [hunk 2]
```diff
@@ -274,6 +274,18 @@ impl ZkWasmCircuitBuilder {
         }
     }
 
+    fn create_params(&self) -> Params<G1Affine>{
+          // Initialize the polynomial commitment parameters
+        let timer = start_timer!(|| format!("build params with K = {}", K));
+        let params: Params<G1Affine> = Params::<G1Affine>::unsafe_setup::<Bn256>(K);
+        end_timer!(timer);
+
```
Comment (junyu0312) [inline src/circuits/mod.rs:282]: Please format modified files.

Comment (nagatoism) [reply src/circuits/mod.rs:283]: done cargo fmt.

File: src/cli/run.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    println!(
+        "read file from {}",
+        fs::canonicalize(&path).unwrap().to_str().unwrap()
+    );
+
+    let mut f = File::open(path).unwrap();
+    f.read_to_end(&mut binary).unwrap();
+
+    fs::create_dir(PathBuf::from(output_path));
```
[the remark this answers is not in the document]
Comment (nagatoism) [reply src/cli/run.rs]: we can abort here if the folder already existed.

File: src/cli/main.rs (hunk 1 above)
Comment (nagatoism) [reply src/cli/main.rs]: removed type argument.

File: src/cli/run.rs [hunk 4]
```diff
@@ -18,36 +19,30 @@ pub struct ArgumentError;
...
+fn parser(f_sig: ExternVal, vv: Vec<&str>) -> Result<Vec<Value>, ArgumentError> {
+    let f_sig = &f_sig.as_func().unwrap().signature();
+    assert_eq!(f_sig.params().len(), vv.len());
+    f_sig
+        .params()
+        .into_iter()
+        .zip(vv.into_iter())
+        .map(|(t, v)| match t {
+            wasmi::ValueType::I32 => Ok(Value::I32(v.parse::<i32>().unwrap())),
+            wasmi::ValueType::I64 => Ok(Value::I32(v.parse::<i32>().unwrap())),
```
Comment (junyu0312) [inline src/cli/run.rs:37]: Ok(Value::I64(v.parse::<i64>().unwrap()))

**Review by lanbones — APPROVED** [review]

**Pull request merged**